src/core.js

```diff
@@ -446,6 +469,86 @@ export default class Editable {
     this.dispatcher.unload()
     return this
   }
+
+  /*
+    Takes coordinates and uses its left value to find out how to offset a character in a string to
+    closely match the coordinates.left value.
+    Takes conditions for the result being on the first line, used when navigating to a paragraph from
+    the above paragraph and being on the last line, used when navigating to a paragraph from the below
+    paragraph.
+
+    Internally this sets up the methods used for a binary cursor search and calls this.
+
+    @param {DomNode} element The DOM Node (usually editable elem) to which the cursor jumps
+    @param {Object} coordinates The bounding rect of the preceeding cursor to be matched
+    @param {Boolean} requiredOnFirstLine set to true if you want to require the cursor to be on the first line of the paragraph
+    @param {Boolean} requiredOnLastLine set to true if you want to require the cursor to be on the last line of the paragraph
+
+    @return {Object} object with boolean `wasFound` indicating if the binary search found an offset and `offset` to indicate the actual character offset
+    */
+  findClosestCursorOffset (
+    {element, origCoordinates, requiredOnFirstLine = false, requiredOnLastLine = false}) {
+    // early terminate on empty editables
+    const totalCharCount = getTotalCharCount(element)
+    if (totalCharCount === 0) return {wasFound: false}
+    // move left if the coordinates are to the left
+    const leftCondition = ({data, coordinates}) => {
+      if (coordinates.left >= data.origCoordinates.left) return true
+      return false
+    }
+    // move up if cursor is required to be on first line but is currently not
+    const upCondition = ({data, cursor}) => {
+      if (data.requiredOnFirstLine && !cursor.isAtFirstLine()) return true
+      return false
+    }
+    // move down if cursor is required to be on last line but is currently not
+    const downCondition = ({data, cursor}) => {
+      if (data.requiredOnLastLine && !cursor.isAtLastLine()) return true
+      return false
+    }
+    // move the binary search index in between the current position and the left limit
+    const moveLeft = (data) => {
+      data.rightLimit = data.currentOffset
+      data.currentOffset = Math.floor((data.currentOffset - data.leftLimit) / 2)
+    }
+    // move the binary search index in between the current position and the right limit
+    const moveRight = (data) => {
+      data.leftLimit = data.currentOffset
+      data.currentOffset = data.currentOffset + Math.floor((data.rightLimit - data.currentOffset) / 2)
+    }
+    // consider a small bluriness since character never exactly match coordinates
+    const foundChecker = ({distance, bluriness}) => {
```

Comment:
I think we should simply return the closest cursor position. There is no need for a `bluriness` factor.